components/brave_vpn/brave_vpn_service.cc

```diff
@@ -0,0 +1,251 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_vpn/brave_vpn_service.h"
+
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+
+namespace {
+constexpr char kVpnHost[] = "housekeeping.sudosecuritygroup.com";
+
+constexpr char kAllServerRegions[] = "api/v1/servers/all-server-regions";
+constexpr char kTimezonesForRegions[] =
+    "api/v1.1/servers/timezones-for-regions";
+constexpr char kHostnameForRegion[] = "api/v1/servers/hostnames-for-region";
+constexpr char kCreateSubscriberCredential[] =
+    "api/v1/subscriber-credential/create";
+constexpr char kVerifyPurchaseToken[] = "api/v1.1/verify-purchase-token";
+
+net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
+  return net::DefineNetworkTrafficAnnotation("brave_vpn_service", R"(
+      semantics {
+        sender: "Brave VPN Service"
+        description:
+          "This service is used to communicate with Guardian VPN apis"
+          "on behalf of the user interacting with the Brave VPN."
+        trigger:
+          "Triggered by user connecting the Brave VPN."
+        data:
+          "Servers, hosts and credentials for Brave VPN"
+        destination: WEBSITE
+      }
+    )");
+}
+
+GURL GetURLWithPath(const std::string& host, const std::string& path) {
+  return GURL(std::string(url::kHttpsScheme) + "://" + host).Resolve(path);
+}
+
+std::string CreateJSONRequestBody(const base::Value& dict) {
+  std::string json;
+  base::JSONWriter::Write(dict, &json);
+  return json;
+}
+
+std::string GetSubscriberCredentialFromJson(const std::string& json) {
+  base::JSONReader::ValueWithError value_with_error =
+      base::JSONReader::ReadAndReturnValueWithError(
+          json, base::JSONParserOptions::JSON_PARSE_RFC);
+  base::Optional<base::Value>& records_v = value_with_error.value;
+  const base::Value* subscriber_credential =
+      records_v->FindKey("subscriber-credential");
```

Comment:
Oops, I missed one more. Need to check `records_v` is dictionary.
If |records_v| is not dictionary, calling `FindKey` on it will give DCHECK failure.